### ISSUE TITLE
CLI app (i.e. `napari.main.main()` ) fail with PNG file, but opens with open menu

### ISSUE THREAD

**Issue by jmetz** [napari/napari#489]
## 🐛 Bug

The CLI (`napari` and the equivalent python call `napari.main.main()`) fail when passing in a PNG or list of PNG files on the command line. 

Opening the same file from the *open* menu has no problems. 

The issue seems to stem from the thumbnailing operation being passed an empty array. 

## To Reproduce

1. Use either `napari file.png` or `python -c "from napari.main import main; main()" file.png`

The traceback is: 

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "napari/main.py", line 38, in main
    v.add_image(image, multichannel=args.multichannel)
  File "napari/components/viewer_model.py", line 391, in add_image
    **kwargs,
  File "napari/layers/image/image.py", line 165, in __init__
    self._set_view_slice()
  File "napari/layers/image/image.py", line 287, in _set_view_slice
    self._update_thumbnail()
  File "napari/layers/image/image.py", line 337, in _update_thumbnail
    self.thumbnail = colormapped
  File "napari/layers/base/base.py", line 301, in thumbnail
    thumbnail = img_as_ubyte(thumbnail)
  File ".../lib/python3.6/site-packages/skimage/util/dtype.py", line 486, in img_as_ubyte
    return convert(image, np.uint8, force_copy)
  File ".../lib/python3.6/site-packages/skimage/util/dtype.py", line 254, in convert
    if np.min(image) < -1.0 or np.max(image) > 1.0:
  File ".../lib/python3.6/site-packages/numpy/core/fromnumeric.py", line 2618, in amin
    initial=initial)
  File ".../lib/python3.6/site-packages/numpy/core/fromnumeric.py", line 86, in _wrapreduction
    return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
ValueError: zero-size array to reduction operation minimum which has no identity
```

## Expected behavior

The file should load as it does with the `open` menu command. 

## Environment

 - napari Version: Reproduced on current head: a33d9ac13d50291b65348753e2093e132b64ec7d and current release
 - OS (e.g., Linux): Linux 4.9.0-9-amd64 #1 SMP Debian 4.9.168-1+deb9u4 (2019-07-19) x86_64 GNU/Linux
 - Python version: Python 3.6.1
 - Any other relevant information: 

file used: 
![media4_frame000000](https://user-images.githubusercontent.com/1950756/63525310-63afad80-c4f5-11e9-94db-c99d67269b65.png)

**Comment by jni** [comment]
@jmetz the key issue is that this png is actually multichannel [1], and the File Open dialog does some multichannel-guessing magic while the command line viewer doesn't. We should unify this behaviour so I'll leave this open, but in the meantime, using `napari -m file.png` works.

.. [1]

```python
In [1]: from skimage import io                                                                                                                                                                                                                                
In [2]: io.imread('test.png').shape                                                                                                                                                                                                                           
Out[2]: (727, 696, 3)  
```

**Comment by jni** [comment]
(Actually the other issue is that we should be able to display a thumbnail for a (696, 3)-shaped image, which is what napari was trying to do. @sofroniewn? =)

**Comment by sofroniewn** [comment]
yup the thumbnail generation has tried scaling that so that it fits inside a 32x32 block - which has led it to scale to a `32x0` image that causes everything to choke - due to the extreme aspect ratio - we should certainly catch this and fix it, i've encountered it before. i'll follow up with a PR that fixes this

**Comment by jmetz** [comment]
Ahh got it - thanks for the clarification @jni . In that case, I guess this is somewhere between a mild bug (the way it fails is still a little ugly) and a feature request that the CLI have the same clever guessing that the internal opener uses.

**Comment by sofroniewn** [comment]
@jni can this now be closed by #560?

**Comment by jni** [comment]
Fixed in #590
